frontend/src/scenes/billing/BillingProduct.tsx

```diff
@@ -82,8 +80,7 @@ export const BillingProduct = ({ product }: { product: BillingProductV2Type }):
     const upgradeToPlanKey = upgradePlan?.plan_key
     const currentPlanKey = currentPlan?.plan_key
     const showUpgradeCard =
-        (upgradePlan?.product_key !== 'platform_and_support' || product?.addons?.length === 0) &&
-        (upgradePlan || (!upgradePlan && !product.current_amount_usd) || (isOnboarding && !product.contact_support))
+        (upgradePlan?.product_key !== 'platform_and_support' || product?.addons?.length === 0) && upgradePlan
```

Comment:
As far as I can tell, the only case when the upgradePlan will be undefined is when the currentPlan is already the highest possible plan.

Reply:
Some of the logic/code in the upgrade card is also based on the current plan (even when there isn't an upgrade plan) to let the user know they are subscribed to that product.

Example: 
![Screenshot 2024-06-06 at 3 28 36 PM](https://github.com/PostHog/posthog/assets/6392501/bb162449-ea5f-4ee3-8381-daa7eb8f3426)

It's pretty buggy, a little confusing and takes up a lot of space, and I agree that it doesn't need to be there, so I'm cool with this. They are going to be removed with my subscription to all products anyway.